raster/src/main/scala/geotrellis/raster/io/geotiff/OverviewStrategy.scala

```diff
@@ -16,20 +16,75 @@
 
 package geotrellis.raster.io.geotiff
 
+import geotrellis.raster.CellSize
+
+import scala.collection.Searching._
+
+/**
+  * Options ported from GDAL (https://gdal.org/programs/gdalwarp.html#cmdoption-gdalwarp-ovr)
+  *  for automatic selection from among available overviews
+  */
 sealed trait OverviewStrategy
+
+object OverviewStrategy {
+  def DEFAULT = Auto(0)
+
+  def selectOverview(
+    overviewCS: List[CellSize],
+    desiredCS: CellSize,
+    strategy: OverviewStrategy
+  ): Int = strategy match {
+    case Level(overviewIdx) =>
+      overviewIdx
+    case Auto(n) =>
+      selectIndexByProximity(overviewCS, desiredCS, 0.5) + n
+    case Base =>
+      overviewCS.indexOf(overviewCS.min)
+    case AutoHigherResolution =>
+      selectIndexByProximity(overviewCS, desiredCS, 1.0)
```

Comment:
Waiting for Travis to be happy with this change. This behavior should match to the tested old behavior (the old one had a long testing path).

Reply:
The tests turned out to require a bit more attention than I'd anticipated as some of them baked in the `AutoHigher` behavior rather than `Auto`